TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/DriveSystem.java

```diff
@@ -162,26 +168,60 @@ public boolean driveToPositionTicks(int ticks, Direction direction, double maxPo
         }
 
         for (DcMotor motor : motors.values()) {
+//            Log.d(TAG, motor.toString() + ", " + motor.getPortNumber() + ": " + motor.getCurrentPosition());
             int offset = Math.abs(motor.getCurrentPosition() - mTargetTicks);
-            if(offset <= 0){
-
+            if(offset <= 15){
                 // Shut down motors
                 setMotorPower(0);
-
                 // Reset motors to default run mode
                 setRunMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
-
                 // Reset target
                 mTargetTicks = 0;
-
                 // Motor has reached target
                 return true;
             }
         }
 
+        double currHeading = imuSystem.getHeading();
+        Log.d("Curr heading: ", currHeading + " ");
+        if (strafeSet && mStrafeHeading != currHeading && Direction.isStrafe(direction)) {
+            double diff = computeDegreesDiff(mStrafeHeading, currHeading);
+            Log.d("Diff: ", diff + " ");
+            double strafe_coeff = direction == Direction.LEFT ? 0.09 : 0.08;
+            double correction = Range.clip(0.09 * diff, -1, 1);
+            int sign = direction == Direction.LEFT ? -1 : 1;
+            motors.forEach((name, motor) -> {
+                switch(name) {
+                    case FRONTLEFT:
+                    case BACKLEFT:
+                        double maxLeft = direction == Direction.LEFT ? 1 : 0.9;
+                        motor.setPower(correction > 0 ? 1 - sign * correction: 1);
+                        //double power = Range.clip(motor.getPower() + (0.23 * diff), -1, 1);
```

Comment:
Are you done with this stuff?

Reply:
I think so? This version of strafing works at least decently well - I thought it would be better to have something which mostly works than that doesn't work at all in master.